src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -289,6 +290,10 @@ public function addCartRuleAction(Request $request, int $cartId): JsonResponse
     {
         $cartRuleId = $request->request->getInt('cartRuleId');
         try {
+            $cartInfo = $this->getCartInfo($cartId)->getShipping();
+            if ($cartInfo->hasFreeShippingCartRule() || $cartInfo->isFreeShipping()) {
+                throw new FreeShippingCartRuleAlreadyExistException('Free shipping cart rule already exist!');
```

Comment:
You should add a translated message matching this exception

Reply:
can you give an example about how to do it. Search through the code and haven't found any translated messages when throwing exceptions.

Reply:
I was referring to the `getErrorMessages` that makes the mapping between exception type and a translated message

Reply:
fixing this one now :)

Reply:
@jolelievre done, ready for review again!